algorithms/integration/processor.py

```diff
@@ -703,21 +700,52 @@ def compute_jobs(self):
             if scan is not None:
                 assert len(imgs) >= len(scan), "Invalid scan range"
                 array_range = scan.get_array_range()
+
             if self.params.block.size is None:
                 block_size_frames = array_range[1] - array_range[0]
+            elif self.params.block.size == libtbx.Auto:
+                # auto determine based on nframes and overlap
+                nframes = array_range[1] - array_range[0]
+                nblocks = self.params.mp.nproc * self.params.mp.njobs
+                # want data to be split into n blocks with overlaps
+                # i.e. [x, overlap, y, overlap, y, overlap, ....,y,  overlap, x]
+                # blocks are x + overlap, or overlap + y + overlap.
+                x = (nframes - block_overlap) / nblocks
+                block_size = int(math.ceil(x + block_overlap))
+                # increase the block size to be at least twice the overlap, in
+                # case the overlap is large e.g. if high mosaicity.
+                block_size_frames = max(block_size, 2 * block_overlap)
             elif self.params.block.units == "radians":
-                phi0, dphi = scan.get_oscillation(deg=False)
+                _, dphi = scan.get_oscillation(deg=False)
                 block_size_frames = int(math.ceil(self.params.block.size / dphi))
+                # if the specified block size is lower than the overlap,
+                # reduce the overlap to be half of the block size.
+                block_overlap = min(
+                    block_overlap, int(math.floor(block_size_frames / 2.0))
+                )
             elif self.params.block.units == "degrees":
-                phi0, dphi = scan.get_oscillation()
+                _, dphi = scan.get_oscillation()
                 block_size_frames = int(math.ceil(self.params.block.size / dphi))
+                # if the specified block size is lower than the overlap,
+                # reduce the overlap to be half of the block size.
+                block_overlap = min(
+                    block_overlap, int(math.floor(block_size_frames / 2.0))
```

Comment:
use the floor divison operator?
```suggestion
                    block_overlap, int(block_size_frames // 2)
```